RecoBTag/FeatureTools/plugins/DeepBoostedJetTagInfoProducer.cc

```diff
@@ -323,10 +326,16 @@ void DeepBoostedJetTagInfoProducer::produce(edm::Event &iEvent, const edm::Event
 float DeepBoostedJetTagInfoProducer::puppiWgt(const reco::CandidatePtr &cand) {
   const auto *pack_cand = dynamic_cast<const pat::PackedCandidate *>(&(*cand));
   const auto *reco_cand = dynamic_cast<const reco::PFCandidate *>(&(*cand));
-  float wgt = 1.;
-  if (pack_cand)
-    wgt = pack_cand->puppiWeight();
-  else if (reco_cand) {
+
+  //
+  // Access puppi weight from ValueMap.
+  //
+  float wgt = 1.;  // Set to fallback value
+
+  if (pack_cand) {
+    if (use_puppi_value_map_)
+      wgt = (*puppi_value_map_)[cand];
```

Comment:
Does this mean that accessing the PUPPI weight from the candidate is not supported anymore? Or can you add a
```
else
    wgt = pack_cand->puppiWeight();
````
here?

Reply:
Here I'd like to ask @hqucms and @nurfikri89 for their feedback, what we already discussed before the PR was submitted is the following, which indeed sounds like accessing the PUPPI weight from the candidate is not supported anymore. The full answer was:

```
Actually, this is not the case anymore but it was for Run-2 releases. In order to be in sync with RECO/AOD level, starting from the Run-3 CMSSW releases (as early as CMSSW_12 I think), 
puppi jet reclustering with MiniAOD input requires to provide packedPFCandidates and the puppi weight ValueMap (produced by the PuppiProducer) to FastJetProducer instead of just the 
new PackedCandidates (produced by the PuppiProducer).

So every time one wants use puppi weights (even if they are the default stored weights in MiniAODs), one has to setup PuppiProducer and then pass the ValueMap to any producers 
that will read in puppi weights. In the case of using the default stored weights, the PuppiProducer has the flag "useExistingWeight" which will enable the PuppiProducer to copy 
the existing weights in MiniAOD into the ValueMap.

[...] this is now the standard practice of reclustering puppi jet with MiniAODs [...]
```

Reply:
Ok, thanks for confirming that